Måske en udjævningsfejl i niv regnemotoren. AssertionError: Mismatch mellem antal koter og varianser

En lille opgave hvor 1 punkt er indmålt mellem 2 kendte koter.
I den endelige beregning hvor begge koter fastholdes kommer denne medd. AssertionError: Mismatch mellem antal koter og varianser.
Nedenfor hele loggen:

```
(fire) F:\GDL\Data\GEO\BC\Niv_Opgaver\2026\KDI_RINGKØBING>fire niv regn 2026_KDI_RINGKØBING
Så regner vi
Fastholder 1 og beregner nye koter for 2 punkter
Skriver: {'Singulære', 'Netgeometri', 'Parametre', 'Kontrolberegning'}
Til filen '2026_KDI_RINGKØBING.xlsx'
Overskriver fanebladene {'Singulære', 'Netgeometri', 'Parametre'}
    med opdaterede versioner.
Foregående versioner beholdes i 'ex'-filen '2026_KDI_RINGKØBING-ex.xlsx'
Færdig! - åbner regneark og resultatrapport for check.

(fire) F:\GDL\Data\GEO\BC\Niv_Opgaver\2026\KDI_RINGKØBING>fire niv regn 2026_KDI_RINGKØBING
Så regner vi
Fastholder 2 og beregner nye koter for 1 punkter
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ in _run_module_as_main:198                                                                       │
│ ╭─────────────────────────────────────────── locals ───────────────────────────────────────────╮ │
│ │   alter_argv = False                                                                         │ │
│ │         code = <code object <module> at 0x0000024AC08AD890, file                             │ │
│ │                "C:\Users\b012858\AppData\Local\mambaforge\envs\fire\Scripts\fire.exe\__main… │ │
│ │                line 1>                                                                       │ │
│ │ main_globals = {                                                                             │ │
│ │                │   '__name__': '__main__',                                                   │ │
│ │                │   '__doc__': None,                                                          │ │
│ │                │   '__package__': '',                                                        │ │
│ │                │   '__loader__': <zipimporter object                                         │ │
│ │                "C:\Users\b012858\AppData\Local\mambaforge\envs\fire\Scripts\fire.exe\">,     │ │
│ │                │   '__spec__': ModuleSpec(name='__main__', loader=<zipimporter object        │ │
│ │                "C:\Users\b012858\AppData\Local\mambaforge\envs\fire\Scripts\fire.exe\">,     │ │
│ │                origin='C:\\Users\\b012858\\AppData\\Local\\mambaforge\\envs\\fire\\Scripts\… │ │
│ │                │   '__annotations__': {},                                                    │ │
│ │                │   '__builtins__': <module 'builtins' (built-in)>,                           │ │
│ │                │   '__file__':                                                               │ │
│ │                'C:\\Users\\b012858\\AppData\\Local\\mambaforge\\envs\\fire\\Scripts\\fire.e… │ │
│ │                │   '__cached__':                                                             │ │
│ │                'C:\\Users\\b012858\\AppData\\Local\\mambaforge\\envs\\fire\\Scripts\\fire.e… │ │
│ │                │   'sys': <module 'sys' (built-in)>,                                         │ │
│ │                │   ... +1                                                                    │ │
│ │                }                                                                             │ │
│ │     mod_name = '__main__'                                                                    │ │
│ │     mod_spec = ModuleSpec(name='__main__', loader=<zipimporter object                        │ │
│ │                "C:\Users\b012858\AppData\Local\mambaforge\envs\fire\Scripts\fire.exe\">,     │ │
│ │                origin='C:\\Users\\b012858\\AppData\\Local\\mambaforge\\envs\\fire\\Scripts\… │ │
│ ╰──────────────────────────────────────────────────────────────────────────────────────────────╯ │
│ in _run_code:88                                                                                  │
│ ╭─────────────────────────────────────────── locals ───────────────────────────────────────────╮ │
│ │       cached = 'C:\\Users\\b012858\\AppData\\Local\\mambaforge\\envs\\fire\\Scripts\\fire.e… │ │
│ │         code = <code object <module> at 0x0000024AC08AD890, file                             │ │
│ │                "C:\Users\b012858\AppData\Local\mambaforge\envs\fire\Scripts\fire.exe\__main… │ │
│ │                line 1>                                                                       │ │
│ │        fname = 'C:\\Users\\b012858\\AppData\\Local\\mambaforge\\envs\\fire\\Scripts\\fire.e… │ │
│ │ init_globals = None                                                                          │ │
│ │       loader = <zipimporter object                                                           │ │
│ │                "C:\Users\b012858\AppData\Local\mambaforge\envs\fire\Scripts\fire.exe\">      │ │
│ │     mod_name = '__main__'                                                                    │ │
│ │     mod_spec = ModuleSpec(name='__main__', loader=<zipimporter object                        │ │
│ │                "C:\Users\b012858\AppData\Local\mambaforge\envs\fire\Scripts\fire.exe\">,     │ │
│ │                origin='C:\\Users\\b012858\\AppData\\Local\\mambaforge\\envs\\fire\\Scripts\… │ │
│ │     pkg_name = ''                                                                            │ │
│ │  run_globals = {                                                                             │ │
│ │                │   '__name__': '__main__',                                                   │ │
│ │                │   '__doc__': None,                                                          │ │
│ │                │   '__package__': '',                                                        │ │
│ │                │   '__loader__': <zipimporter object                                         │ │
│ │                "C:\Users\b012858\AppData\Local\mambaforge\envs\fire\Scripts\fire.exe\">,     │ │
│ │                │   '__spec__': ModuleSpec(name='__main__', loader=<zipimporter object        │ │
│ │                "C:\Users\b012858\AppData\Local\mambaforge\envs\fire\Scripts\fire.exe\">,     │ │
│ │                origin='C:\\Users\\b012858\\AppData\\Local\\mambaforge\\envs\\fire\\Scripts\… │ │
│ │                │   '__annotations__': {},                                                    │ │
│ │                │   '__builtins__': <module 'builtins' (built-in)>,                           │ │
│ │                │   '__file__':                                                               │ │
│ │                'C:\\Users\\b012858\\AppData\\Local\\mambaforge\\envs\\fire\\Scripts\\fire.e… │ │
│ │                │   '__cached__':                                                             │ │
│ │                'C:\\Users\\b012858\\AppData\\Local\\mambaforge\\envs\\fire\\Scripts\\fire.e… │ │
│ │                │   'sys': <module 'sys' (built-in)>,                                         │ │
│ │                │   ... +1                                                                    │ │
│ │                }                                                                             │ │
│ │  script_name = None                                                                          │ │
│ ╰──────────────────────────────────────────────────────────────────────────────────────────────╯ │
│                                                                                                  │
│ in <module>:6                                                                                    │
│                                                                                                  │
│ C:\Users\b012858\AppData\Local\mambaforge\envs\fire\Lib\site-packages\click\core.py:1442 in      │
│ __call__                                                                                         │
│                                                                                                  │
│ C:\Users\b012858\AppData\Local\mambaforge\envs\fire\Lib\site-packages\click\core.py:1363 in main │
│                                                                                                  │
│ C:\Users\b012858\AppData\Local\mambaforge\envs\fire\Lib\site-packages\click\core.py:1830 in      │
│ invoke                                                                                           │
│                                                                                                  │
│ C:\Users\b012858\AppData\Local\mambaforge\envs\fire\Lib\site-packages\click\core.py:1830 in      │
│ invoke                                                                                           │
│                                                                                                  │
│ C:\Users\b012858\AppData\Local\mambaforge\envs\fire\Lib\site-packages\click\core.py:1226 in      │
│ invoke                                                                                           │
│                                                                                                  │
│ C:\Users\b012858\AppData\Local\mambaforge\envs\fire\Lib\site-packages\click\core.py:794 in       │
│ invoke                                                                                           │
│                                                                                                  │
│ C:\FIRE\src\fire\cli\niv\_regn.py:317 in regn                                                    │
│                                                                                                  │
│   314 │   )                                                                                      │
│   315 │                                                                                          │
│   316 │   try:                                                                                   │
│ ❱ 317 │   │   motor.udjævn()                                                                     │
│   318 │   except UdjævningFejl as fejl:                                                          │
│   319 │   │   fire.cli.print(                                                                    │
│   320 │   │   │   f"FEJL: {fejl}",                                                               │
│                                                                                                  │
│ ╭─────────────────────────────────────────── locals ───────────────────────────────────────────╮ │
│ │            aktuelt_faneblad = 'Kontrolberegning'                                             │ │
│ │                  arbejdssæt = │   │   Punkt Fasthold             Hvornår      Kote         σ │ │
│ │                               ...  System       Nord       Øst  uuid Udelad publikation      │ │
│ │                               0  K-73-09001        x                 NaT  2.636290  0.519749 │ │
│ │                               ...   DVR90  56.089791  8.238611                               │ │
│ │                               1  K-73-09025          2024-03-27 10:27:00       NaN       NaN │ │
│ │                               ...   DVR90  56.088112  8.238555                               │ │
│ │                               2  K-73-09115        x 2024-03-27 10:27:00  3.370219  0.363665 │ │
│ │                               ...   DVR90  56.088656  8.241052                               │ │
│ │                                                                                              │ │
│ │                               [3 rows x 14 columns]                                          │ │
│ │         beregningsparametre = {'regnemotor': 'GamaRegn'}                                     │ │
│ │              beregningstype = 'endelig'                                                      │ │
│ │              ensomme_subnet = []                                                             │ │
│ │         estimerbare_punkter = ['K-73-09025']                                                 │ │
│ │            findes_parameter = True                                                           │ │
│ │                      infiks = ''                                                             │ │
│ │                     kontrol = False                                                          │ │
│ │                kontrolværdi = 'GamaRegn'                                                     │ │
│ │                      kwargs = {'db': None, 'monokrom': None, 'debug': None}                  │ │
│ │                       motor = <fire.api.niv.regnemotor.GamaRegn object at                    │ │
│ │                               0x0000024ACF9A16A0>                                            │ │
│ │                 motorkwargs = {}                                                             │ │
│ │            net_uden_ensomme = {                                                              │ │
│ │                               │   'K-73-09001': ['K-73-09025'],                              │ │
│ │                               │   'K-73-09025': ['K-73-09001', 'K-73-09115'],                │ │
│ │                               │   'K-73-09115': ['K-73-09025']                               │ │
│ │                               }                                                              │ │
│ │              næste_faneblad = 'Endelig beregning'                                            │ │
│ │               observationer = │   Journal Sluk         Fra         Til       ΔH       L      │ │
│ │                               Opst  ...  Sky  Sol Vind Sigt               Kilde  Type  uuid  │ │
│ │                               0  271370:1       K-73-09001  K-73-09025 -1.37719  248.71      │ │
│ │                               5  ...    7    2    2  200  kdi_ringkøbing.txt   MTL           │ │
│ │                               1  271370:2       K-73-09025  K-73-09001  1.37658  248.23      │ │
│ │                               5  ...    0    0    0    0  kdi_ringkøbing.txt   MTL           │ │
│ │                               2  271371:1       K-73-09025  K-73-09115  2.11112  254.34      │ │
│ │                               5  ...    0    0    0    0  kdi_ringkøbing.txt   MTL           │ │
│ │                               3  271371:2       K-73-09115  K-73-09025 -2.11102  254.53      │ │
│ │                               5  ...    0    0    0    0  kdi_ringkøbing.txt   MTL           │ │
│ │                                                                                              │ │
│ │                               [4 rows x 19 columns]                                          │ │
│ │ observationer_uden_slukkede = │   Journal Sluk         Fra         Til       ΔH       L      │ │
│ │                               Opst  ...  Sky  Sol Vind Sigt               Kilde  Type  uuid  │ │
│ │                               0  271370:1       K-73-09001  K-73-09025 -1.37719  248.71      │ │
│ │                               5  ...    7    2    2  200  kdi_ringkøbing.txt   MTL           │ │
│ │                               1  271370:2       K-73-09025  K-73-09001  1.37658  248.23      │ │
│ │                               5  ...    0    0    0    0  kdi_ringkøbing.txt   MTL           │ │
│ │                               2  271371:1       K-73-09025  K-73-09115  2.11112  254.34      │ │
│ │                               5  ...    0    0    0    0  kdi_ringkøbing.txt   MTL           │ │
│ │                               3  271371:2       K-73-09115  K-73-09025 -2.11102  254.53      │ │
│ │                               5  ...    0    0    0    0  kdi_ringkøbing.txt   MTL           │ │
│ │                                                                                              │ │
│ │                               [4 rows x 19 columns]                                          │ │
│ │                   parameter = 'regnemotor'                                                   │ │
│ │                   parametre = │   │    Navn     Værdi                                        │ │
│ │                               0     Version    1.10.0                                        │ │
│ │                               1    Database      prod                                        │ │
│ │                               2  regnemotor  GamaRegn                                        │ │
│ │                        plot = False                                                          │ │
│ │                 projektnavn = '2026_KDI_RINGKØBING'                                          │ │
│ │               punktoversigt = │   │   Punkt Fasthold             Hvornår      Kote         σ │ │
│ │                               ...  System       Nord       Øst  uuid Udelad publikation      │ │
│ │                               0  K-73-09025                          NaT       NaN       NaN │ │
│ │                               ...   DVR90  56.088112  8.238555                               │ │
│ │                               1  K-73-09001          2024-03-27 10:27:00  2.636290  0.519749 │ │
│ │                               ...   DVR90  56.089791  8.238611                               │ │
│ │                               2  K-73-09115        x 2024-03-27 10:27:00  3.370219  0.363665 │ │
│ │                               ...   DVR90  56.088656  8.241052                               │ │
│ │                                                                                              │ │
│ │                               [3 rows x 14 columns]                                          │ │
│ │              regneparametre = ()                                                             │ │
│ │                  resultater = {                                                              │ │
│ │                               │   'Netgeometri':         Punkt      Nabo 0      Nabo 1       │ │
│ │                               0  K-73-09001  K-73-09025                                      │ │
│ │                               1  K-73-09025  K-73-09001  K-73-09115                          │ │
│ │                               2  K-73-09115  K-73-09025            ,                         │ │
│ │                               │   'Singulære': Empty DataFrame                               │ │
│ │                               Columns: [Punkt]                                               │ │
│ │                               Index: []                                                      │ │
│ │                               }                                                              │ │
│ │                       værdi = 'GamaRegn'                                                     │ │
│ ╰──────────────────────────────────────────────────────────────────────────────────────────────╯ │
│                                                                                                  │
│ C:\FIRE\src\fire\api\niv\regnemotor.py:508 in udjævn                                             │
│                                                                                                  │
│   505 │   │                                                                                      │
│   506 │   │   self.skriv_gama_inputfil()                                                         │
│   507 │   │   self.kald_gama()                                                                   │
│ ❱ 508 │   │   self.nye_koter = self.læs_gama_outputfil()                                         │
│   509                                                                                            │
│   510                                                                                            │
│   511 class DumRegn(RegneMotor):                                                                 │
│                                                                                                  │
│ ╭──────────────────────────────── locals ────────────────────────────────╮                       │
│ │ self = <fire.api.niv.regnemotor.GamaRegn object at 0x0000024ACF9A16A0> │                       │
│ ╰────────────────────────────────────────────────────────────────────────╯                       │
│                                                                                                  │
│ C:\FIRE\src\fire\api\niv\regnemotor.py:486 in læs_gama_outputfil                                 │
│                                                                                                  │
│   483 │   │   if isinstance(varliste, dict):                                                     │
│   484 │   │   │   varliste = [varliste]                                                          │
│   485 │   │                                                                                      │
│ ❱ 486 │   │   assert len(koteliste) == len(                                                      │
│   487 │   │   │   varliste                                                                       │
│   488 │   │   ), "Mismatch mellem antal koter og varianser"                                      │
│   489                                                                                            │
│                                                                                                  │
│ ╭─────────────────────────────────────────── locals ───────────────────────────────────────────╮ │
│ │       doc = {                                                                                │ │
│ │             │   'gama-local-adjustment': {                                                   │ │
│ │             │   │   '@xmlns': 'http://www.gnu.org/software/gama/gama-local-adjustment',      │ │
│ │             │   │   'description': 'Nivellementsprojekt "2026_KDI_RINGK\\xd8BING"',          │ │
│ │             │   │   'network-general-parameters': {                                          │ │
│ │             │   │   │   '@gama-local-version': '2.19',                                       │ │
│ │             │   │   │   '@gama-local-algorithm': 'gso',                                      │ │
│ │             │   │   │   '@gama-local-compiler': 'Visual C++ 19.16',                          │ │
│ │             │   │   │   '@axes-xy': 'en',                                                    │ │
│ │             │   │   │   '@angles': 'left-handed',                                            │ │
│ │             │   │   │   '@epoch': '0.0000000',                                               │ │
│ │             │   │   │   '@latitude': '55.7000000',                                           │ │
│ │             │   │   │   '@ellipsoid': 'grs80'                                                │ │
│ │             │   │   },                                                                       │ │
│ │             │   │   'network-processing-summary': {                                          │ │
│ │             │   │   │   'coordinates-summary': {                                             │ │
│ │             │   │   │   │   'coordinates-summary-adjusted': {                                │ │
│ │             │   │   │   │   │   'count-xyz': '0',                                            │ │
│ │             │   │   │   │   │   'count-xy': '0',                                             │ │
│ │             │   │   │   │   │   'count-z': '1'                                               │ │
│ │             │   │   │   │   },                                                               │ │
│ │             │   │   │   │   'coordinates-summary-constrained': {                             │ │
│ │             │   │   │   │   │   'count-xyz': '0',                                            │ │
│ │             │   │   │   │   │   'count-xy': '0',                                             │ │
│ │             │   │   │   │   │   'count-z': '0'                                               │ │
│ │             │   │   │   │   },                                                               │ │
│ │             │   │   │   │   'coordinates-summary-fixed': {                                   │ │
│ │             │   │   │   │   │   'count-xyz': '0',                                            │ │
│ │             │   │   │   │   │   'count-xy': '0',                                             │ │
│ │             │   │   │   │   │   'count-z': '2'                                               │ │
│ │             │   │   │   │   }                                                                │ │
│ │             │   │   │   },                                                                   │ │
│ │             │   │   │   'observations-summary': {                                            │ │
│ │             │   │   │   │   'distances': '0',                                                │ │
│ │             │   │   │   │   'directions': '0',                                               │ │
│ │             │   │   │   │   'angles': '0',                                                   │ │
│ │             │   │   │   │   'xyz-coords': '0',                                               │ │
│ │             │   │   │   │   'h-diffs': '4',                                                  │ │
│ │             │   │   │   │   'z-angles': '0',                                                 │ │
│ │             │   │   │   │   's-dists': '0',                                                  │ │
│ │             │   │   │   │   'vectors': '0',                                                  │ │
│ │             │   │   │   │   'azimuths': '0'                                                  │ │
│ │             │   │   │   },                                                                   │ │
│ │             │   │   │   'project-equations': {                                               │ │
│ │             │   │   │   │   'equations': '4',                                                │ │
│ │             │   │   │   │   'unknowns': '1',                                                 │ │
│ │             │   │   │   │   'degrees-of-freedom': '3',                                       │ │
│ │             │   │   │   │   'defect': '0',                                                   │ │
│ │             │   │   │   │   'sum-of-squares': '1.3516108e+00',                               │ │
│ │             │   │   │   │   'connected-network': None                                        │ │
│ │             │   │   │   },                                                                   │ │
│ │             │   │   │   'standard-deviation': {                                              │ │
│ │             │   │   │   │   'apriori': '1.0000000e+00',                                      │ │
│ │             │   │   │   │   'aposteriori': '6.7122047e-01',                                  │ │
│ │             │   │   │   │   'used': 'aposteriori',                                           │ │
│ │             │   │   │   │   'probability': '0.950',                                          │ │
│ │             │   │   │   │   'ratio': '0.671',                                                │ │
│ │             │   │   │   │   'lower': '0.268',                                                │ │
│ │             │   │   │   │   'upper': '1.765',                                                │ │
│ │             │   │   │   │   'passed': None,                                                  │ │
│ │             │   │   │   │   'confidence-scale': '3.1824491e+00'                              │ │
│ │             │   │   │   }                                                                    │ │
│ │             │   │   },                                                                       │ │
│ │             │   │   'coordinates': {                                                         │ │
│ │             │   │   │   'fixed': {                                                           │ │
│ │             │   │   │   │   'point': [                                                       │ │
│ │             │   │   │   │   │   {'id': 'K-73-09001', 'z': '2.636290'},                       │ │
│ │             │   │   │   │   │   {'id': 'K-73-09115', 'z': '3.370219'}                        │ │
│ │             │   │   │   │   ]                                                                │ │
│ │             │   │   │   },                                                                   │ │
│ │             │   │   │   'approximate': {'point': {'id': 'K-73-09025', 'z': '1.259100'}},     │ │
│ │             │   │   │   'adjusted': {                                                        │ │
│ │             │   │   │   │   'point': {'id': 'K-73-09025', 'z': '1.2592793462458056'}         │ │
│ │             │   │   │   },                                                                   │ │
│ │             │   │   │   'orientation-shifts': None,                                          │ │
│ │             │   │   │   'cov-mat': {'dim': '1', 'band': '0', 'flt': '2.1651308e-02'},        │ │
│ │             │   │   │   'original-index': {'ind': '1'}                                       │ │
│ │             │   │   },                                                                       │ │
│ │             │   │   'observations': {                                                        │ │
│ │             │   │   │   'height-diff': [                                                     │ │
│ │             │   │   │   │   {                                                                │ │
│ │             │   │   │   │   │   '@extern': '271370:1',                                       │ │
│ │             │   │   │   │   │   'from': 'K-73-09001',                                        │ │
│ │             │   │   │   │   │   'to': 'K-73-09025',                                          │ │
│ │             │   │   │   │   │   'obs': '-1.3771899999999999',                                │ │
│ │             │   │   │   │   │   'adj': '-1.3770103889334133',                                │ │
│ │             │   │   │   │   │   'stdev': '0.1471438330820646',                               │ │
│ │             │   │   │   │   │   'qrr': '0.141',                                              │ │
│ │             │   │   │   │   │   'f': '49.599',                                               │ │
│ │             │   │   │   │   │   'std-residual': '0.712',                                     │ │
│ │             │   │   │   │   │   'err-obs': '0.241',                                          │ │
│ │             │   │   │   │   │   ... +1                                                       │ │
│ │             │   │   │   │   },                                                               │ │
│ │             │   │   │   │   {                                                                │ │
│ │             │   │   │   │   │   '@extern': '271370:2',                                       │ │
│ │             │   │   │   │   │   'from': 'K-73-09025',                                        │ │
│ │             │   │   │   │   │   'to': 'K-73-09001',                                          │ │
│ │             │   │   │   │   │   'obs': '1.3765799999999999',                                 │ │
│ │             │   │   │   │   │   'adj': '1.3770103889334133',                                 │ │
│ │             │   │   │   │   │   'stdev': '0.1471438330820646',                               │ │
│ │             │   │   │   │   │   'qrr': '0.141',                                              │ │
│ │             │   │   │   │   │   'f': '49.527',                                               │ │
│ │             │   │   │   │   │   'std-residual': '1.710',                                     │ │
│ │             │   │   │   │   │   'err-obs': '0.578',                                          │ │
│ │             │   │   │   │   │   ... +1                                                       │ │
│ │             │   │   │   │   },                                                               │ │
│ │             │   │   │   │   {                                                                │ │
│ │             │   │   │   │   │   '@extern': '271371:1',                                       │ │
│ │             │   │   │   │   │   'from': 'K-73-09025',                                        │ │
│ │             │   │   │   │   │   'to': 'K-73-09115',                                          │ │
│ │             │   │   │   │   │   'obs': '2.1111200000000001',                                 │ │
│ │             │   │   │   │   │   'adj': '2.1109397093933180',                                 │ │
│ │             │   │   │   │   │   'stdev': '0.1471438330820646',                               │ │
│ │             │   │   │   │   │   'qrr': '0.147',                                              │ │
│ │             │   │   │   │   │   'f': '50.427',                                               │ │
│ │             │   │   │   │   │   'std-residual': '0.699',                                     │ │
│ │             │   │   │   │   │   'err-obs': '-0.239',                                         │ │
│ │             │   │   │   │   │   ... +1                                                       │ │
│ │             │   │   │   │   },                                                               │ │
│ │             │   │   │   │   {                                                                │ │
│ │             │   │   │   │   │   '@extern': '271371:2',                                       │ │
│ │             │   │   │   │   │   'from': 'K-73-09115',                                        │ │
│ │             │   │   │   │   │   'to': 'K-73-09025',                                          │ │
│ │             │   │   │   │   │   'obs': '-2.1110199999999999',                                │ │
│ │             │   │   │   │   │   'adj': '-2.1109397093933180',                                │ │
│ │             │   │   │   │   │   'stdev': '0.1471438330820646',                               │ │
│ │             │   │   │   │   │   'qrr': '0.148',                                              │ │
│ │             │   │   │   │   │   'f': '50.455',                                               │ │
│ │             │   │   │   │   │   'std-residual': '0.311',                                     │ │
│ │             │   │   │   │   │   'err-obs': '0.106',                                          │ │
│ │             │   │   │   │   │   ... +1                                                       │ │
│ │             │   │   │   │   }                                                                │ │
│ │             │   │   │   ]                                                                    │ │
│ │             │   │   }                                                                        │ │
│ │             │   }                                                                            │ │
│ │             }                                                                                │ │
│ │ koteliste = [{'id': 'K-73-09025', 'z': '1.2592793462458056'}]                                │ │
│ │  resultat = <_io.TextIOWrapper name='2026_KDI_RINGKØBING-resultat-endelig.xml' mode='r'      │ │
│ │             encoding='cp1252'>                                                               │ │
│ │      self = <fire.api.niv.regnemotor.GamaRegn object at 0x0000024ACF9A16A0>                  │ │
│ │  varliste = '2.1651308e-02'                                                                  │ │
│ ╰──────────────────────────────────────────────────────────────────────────────────────────────╯ │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
AssertionError: Mismatch mellem antal koter og varianser
```

(fire) F:\GDL\Data\GEO\BC\Niv_Opgaver\2026\KDI_RINGKØBING>
@kbevers @krebslw

--- a/src/fire/api/niv/regnemotor.py
+++ b/src/fire/api/niv/regnemotor.py
@@ -478,9 +478,9 @@ class GamaRegn(RegneMotor):
         varliste = doc["gama-local-adjustment"]["coordinates"]["cov-mat"]["flt"]
 
         # Konverter til liste i tilfælde af der kun er blevet udjævnet ét punkt.
-        if isinstance(koteliste, dict):
+        if not isinstance(koteliste, list):
             koteliste = [koteliste]
-        if isinstance(varliste, dict):
+        if not isinstance(varliste, list):
             varliste = [varliste]
 
         assert len(koteliste) == len(

